### PR TITLE
feat: add builder pattern for eval requests for java sdk

### PR DESCRIPTION
**Pull request by markphelps**
🚧

### CODE REVIEW

File: flipt-java/src/main/java/io/flipt/api/evaluation/models/BatchEvaluationRequest.java [hunk 1]
```diff
@@ -32,4 +32,35 @@ public Optional<String> getRequestId() {
...
+
+    public BatchEvaluationRequestBuilder requests(List<EvaluationRequest> requests) {
+      this.requests = requests;
+      return this;
+    }
+
+    public BatchEvaluationRequestBuilder reference(Optional<String> reference) {
+      this.reference = reference;
+      return this;
+    }
```
Comment (GeorgeMac) [inline flipt-java/src/main/java/io/flipt/api/evaluation/models/BatchEvaluationRequest.java:60]:
```suggestion
    public BatchEvaluationRequestBuilder reference(String reference) {
      this.reference = Optional.of(reference);
      return this;
    }
```

File: flipt-java/src/main/java/io/flipt/api/evaluation/models/BatchEvaluationRequest.java [hunk 2]
```diff
@@ -32,4 +32,35 @@ public Optional<String> getRequestId() {
...
+
+  public static final class BatchEvaluationRequestBuilder {
+    private Optional<String> requestId;
+    private List<EvaluationRequest> requests;
+    private Optional<String> reference;
+
+    public BatchEvaluationRequestBuilder() {}
+
+    public BatchEvaluationRequestBuilder requestId(Optional<String> requestId) {
+      this.requestId = requestId;
```
Comment (GeorgeMac) [inline flipt-java/src/main/java/io/flipt/api/evaluation/models/BatchEvaluationRequest.java:48]:
```suggestion
    public BatchEvaluationRequestBuilder requestId(String requestId) {
      this.requestId = Optional.of(requestId);
```

File: flipt-java/src/main/java/io/flipt/api/evaluation/models/EvaluationRequest.java [hunk 3]
```diff
@@ -50,4 +50,47 @@ public Map<String, String> getContext() {
...
+      return this;
+    }
+
+    public EvaluationRequestBuilder context(Map<String, String> context) {
+      this.context = context;
+      return this;
+    }
+
+    public EvaluationRequestBuilder reference(Optional<String> reference) {
+      this.reference = reference;
```
Comment (GeorgeMac) [inline flipt-java/src/main/java/io/flipt/api/evaluation/models/EvaluationRequest.java:88]:
```suggestion
    public EvaluationRequestBuilder reference(String reference) {
      this.reference = Optional.of(reference);
```

**Review by GeorgeMac — COMMENTED** [review]
Some suggestions, but otherwise looks great 👍

**Review by yquansah — APPROVED** [review]

**Pull request merged**